src/expr.rs

```diff
@@ -549,6 +550,21 @@ pub mod parsing {
         ))
     ));
 
+    named!(expr_loop -> Expr, do_parse!(
+        id: option!(
+            terminated!(
+                preceded!(
+                    punct!("'"),
+                    ident),
+                punct!(":"))) >>
+            punct!("loop") >>
+        loop_block: block >>
+        (Expr::Loop(
+            Box::new(loop_block),
+            id,
```

Comment:
I would like the single quote `'` to be part of the id here. That's how Rust's AST does it.

Here is what I did to check. Created `main.rs`:

```rust
fn main() {
    'label: loop {}
}
```

Ran `rustc -Z ast-json main.rs`. The following appears in the output:

```json
"node": {
    "variant": "Loop",
    "fields": [
    {
        "stmts": [],
        "id": 7,
        "rules": "Default",
        "span": {
        "lo": 29,
        "hi": 31
        }
    },
    {
        "node": "'label",
        "span": {
        "lo": 16,
        "hi": 22
        }
    }
    ]
},
```

The label is `"'label"` not just `"label"`.